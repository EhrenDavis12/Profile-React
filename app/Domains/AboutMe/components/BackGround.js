import PropTypes from 'prop-types';
import React from 'react';
import { Parallax } from 'react-parallax';

import BkgImgCalm from 'images/background-calm-clouds.jpg';

// example of Parallax
// https://codesandbox.io/embed/r0yEkozrw?view=preview

function BackGround(props) {
  return (
    <Parallax bgImage={BkgImgCalm} strength={props.strength || 500}>
      <div className="container">{props.children}</div>
    </Parallax>
  );
}

BackGround.propTypes = {
  children: PropTypes.node.isRequired,
  strength: PropTypes.number,
};

export default BackGround;
