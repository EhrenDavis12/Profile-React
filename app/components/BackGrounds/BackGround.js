import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';
import ThemeFactory from 'components/SharedStyles/ThemeFactory';

const theme = ThemeFactory();

const BkgDiv = styled.div`
  background-color: ${theme.PrimaryBkg};
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
`;

// example of Parallax
// https://codesandbox.io/embed/r0yEkozrw?view=preview

function BackGround(props) {
  return (
    // <Parallax bgImage={BkgImgCalm} strength={props.strength || 500}>
    <BkgDiv>{props.children}</BkgDiv>
    // </Parallax>
  );
}

BackGround.propTypes = {
  children: PropTypes.node.isRequired,
  // strength: PropTypes.number,
};

export default BackGround;
