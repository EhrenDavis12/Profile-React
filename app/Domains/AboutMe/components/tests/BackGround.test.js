import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import BackGround from '../BackGround';

describe('<BackGround />', () => {
  it('should render the page', () => {
    const renderedComponent = shallow(
      <BackGround>
        <div>Hello Wold</div>
      </BackGround>,
    );
    expect(renderedComponent).toMatchSnapshot();
  });
});
