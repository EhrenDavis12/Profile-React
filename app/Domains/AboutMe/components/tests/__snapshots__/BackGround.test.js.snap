// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackGround /> should render the page 1`] = `
<t
  bgClassName="react-parallax-bgimage"
  bgImage="IMAGE_MOCK"
  bgImageAlt=""
  className=""
  contentClassName=""
  disabled={false}
  log={false}
  strength={500}
>
  <div
    className="container"
  >
    <div>
      Hello Wold
    </div>
  </div>
</t>
`;
